### PR TITLE
Added logging guide

### PR DESCRIPTION
**Pull request by lucasvickers**
Finally finished the logging guide.  Let us all bask in its glory.

### CODE REVIEW

File: docs/htmlsrc/guides/logging/index.html [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	<body id="guide-contents" class="language-c++">
+
+		<!-- CONTENT STARTS HERE -->
+		<h1>Logging in Cinder</h1>
+        
+        <p>
+        	This document provides an overview of the logging capabilities in Cinder.  You can use this, along with the sample in <em><a href="https://github.com/cinder/Cinder/tree/master/samples/Logging" target="_blank">samples/Logging</a></em>, to quickly begin utilizing Cinder's built in logging capabilities.
+        </p>
+        <a name="overview"></a>
+        <h2>Overview</h2>
```
Comment (gregkepler) [inline docs/htmlsrc/guides/logging/index.html]: Each section within the page should be within <section> tags, so wherever a new h2 starts or so.

File: docs/htmlsrc/guides/logging/index.html [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        <tr><td>LEVEL_INFO</td><td>CI_LOG_I</td><td>2</td><td></td></tr> 
+        <tr><td>LEVEL_WARNING</td><td>CI_LOG_W</td><td>3</td><td></td></tr> 
+        <tr><td>LEVEL_ERROR</td><td>CI_LOG_E</td><td>4</td><td></td></tr> 
+        <tr><td>LEVEL_FATAL</td><td>CI_LOG_F</td><td>5</td><td>(highest logging level)</td></tr> 
+        </table>
+		<br>
+		<a name="types"></a>
+        <h2>Logger Types</h2>
+
+        <h3>Logger</h3>
```
Comment (gregkepler) [inline docs/htmlsrc/guides/logging/index.html]: Perhaps all of theses <h3> elements can be list items

File: docs/htmlsrc/guides/logging/index.html [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        <h3>LoggerFile</h3>
+        <ci>ci::log::LoggerFile</ci> directs log messages to a specified file.<br>
+        <br>
+        <h3>LoggerFileRotating</h3>
+        <ci>ci::log::LoggerFileRotating</ci> directs log messages to a specified file.  The name of that file is evaluated at the first logging call after application startup, and re-evaluated at the first logging call after midnight.<br>
+        <br>
+	    <h3>LoggerSystem</h3>
+        <ci>ci::log::LoggerSystem</ci> will write log messages to the platform specific system log.  This currently means the system log on OS X <code>/var/log/system.log</code> and the Event Log on Windows with MSW applications.  WinRT system logging is still being developed.<br>
+        <br>
+        <h4>Minimum System Logging Level</h4>
```
Comment (gregkepler) [inline docs/htmlsrc/guides/logging/index.html]: if the h3s are turned into list items, then these h4s can be h3s

File: docs/htmlsrc/guides/logging/index.html [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        <table>
+        <tr><td><b>Cinder Level</b></td><td><b>Event Log Level</b></td></tr> 
+        <tr><td>LEVEL_VERBOSE</td><td>EVENTLOG_INFORMATION_TYPE</td></tr> 
+        <tr><td>LEVEL_DEBUG</td><td>EVENTLOG_INFORMATION_TYPE</td></tr> 
+        <tr><td>LEVEL_INFO</td><td>EVENTLOG_INFORMATION_TYPE</td></tr> 
+        <tr><td>LEVEL_WARNING</td><td>EVENTLOG_WARNING_TYPE</td></tr> 
+        <tr><td>LEVEL_ERROR</td><td>EVENTLOG_ERROR_TYPE</td></tr> 
+        <tr><td>LEVEL_FATAL</td><td>EVENTLOG_ERROR_TYPE</td></tr> 
+        </table>
+        <b>Windows:</b> For deployments on Windows systems, you may find that you need to create and register an application manifest file.  We are exploring how this may be integrated into the Cinder workflow, but in the mean time, please reference issue <a href="https://github.com/cinder/Cinder/issues/1109" target="_blank">#1109</a> for information.<br>
```
Comment (gregkepler) [inline docs/htmlsrc/guides/logging/index.html]: Is it weird that there are 2 "Windows:"? Maybe this one should be "Note:"?

File: docs/htmlsrc/guides/logging/index.html [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+		<!-- CONTENT STARTS HERE -->
+		<h1>Logging in Cinder</h1>
+        
+        <p>
+        	This document provides an overview of the logging capabilities in Cinder.  You can use this, along with the sample in <em><a href="https://github.com/cinder/Cinder/tree/master/samples/Logging" target="_blank">samples/Logging</a></em>, to quickly begin utilizing Cinder's built in logging capabilities.
+        </p>
+        <a name="overview"></a>
+        <h2>Overview</h2>
+        Cinder logging resides in the <ci>ci::log</ci> namespace, and provides a set of tools for straight forward, and relatively flexible logging.  The purpose of Cinder logging is not to circumvent advanced logging packages, but rather to provide Cinder users with basic, integrated functionality that will cover most practical use cases.<br>
```
Comment (gregkepler) [inline docs/htmlsrc/guides/logging/index.html]: Paragraphs should be wrapped in `<p></p>`

**Comment by gregkepler** [comment]
@lucasvickers I made a few suggestions. It's mostly just nitpicky semantic html stuff. But otherwise looks great.

**Comment by lucasvickers** [comment]
Thanks for the comments.  I've integrated the changes and it should be ready for merge (pending your review).

**Comment by gregkepler** [comment]
Sweet. Looks good to me.

**Comment by pizthewiz** [comment]
Should this get merged into the 0.9.0 branch too so that when the docs get bumped users will be able to see it? Assuming it doesn't have any functional changes to the `Logger` stuff…

**Comment by lucasvickers** [comment]
This is based on the 0.9.0 code, so this is applicable to that release.

**Comment by richardeakin** [comment]
Can do that. Other question I have, is some link for this guide needed for it to show up on [this page](https://libcinder.org/docs/branch/master/guides/index.html)?

**Comment by richardeakin** [comment]
Hm, I was looking at creating a new PR that merged @lucasvickers's logging-guide branch into release_v0.9.0:

https://github.com/cinder/Cinder/compare/release_v0.9.0...lucasvickers:logging-guide?expand=1

But it looks like it will modify the project version numbers to be 0.9.1dev, which we don't want in that branch. Same with logging-html branch:

https://github.com/cinder/Cinder/compare/release_v0.9.0...lucasvickers:logging-html?expand=1

Happy to try to get this into the release_v0.9.0 branch so it appears on the website (@gregkepler is that how it works?), but don't want to mess up these version strings.

**Pull request merged**